pkg/providers/aws/provider_redis.go

```diff
@@ -99,6 +101,34 @@ func (p *AWSRedisProvider) CreateRedis(ctx context.Context, r *v1alpha1.Redis) (
 	// setup aws elasticache cluster sdk session
 	cacheSvc, stsSvc := createAWSService(stratCfg, providerCreds)
 
+	// Retrieve service maintenance updates, create and export Prometheus metrics
+	output, err := cacheSvc.DescribeServiceUpdates(&elasticache.DescribeServiceUpdatesInput{})
+	if err != nil {
+		logrus.Info(fmt.Sprintf("elasticache serviceupdates error: %s:", err))
+	} else {
+		logrus.Info(fmt.Sprintf("there are elasticache serviceupdates: %s available", len(output.ServiceUpdates)))
+		metricName := "cro_aws_elasticache_service_maintenance"
```

Comment:
These metric names, I think we can move them back to `custom_metrics.go` as a const, and just export them. So we can reference them `resources.PostgresMaintenanceMetricName`, at least then we can keep all metric names together

Reply:
I'd prefer to stick to the interface set by the: `SetMetric(name string, labels map[string]string, epochTimestamp int64)`

We can go with a const in `pkg/providers/aws/provider_redis.go` though?

Reply:
Yeah either way I am not that pushed, my thinking was trying to keep as much metric related stuff and things out of the providers/controllers to avoid bloat. But since its a var name, i have not strong opinion :)